Source/NSOrderedSet.m

```diff
@@ -0,0 +1,1814 @@
+  id object = nil;
+  NSMutableSet *results = [NSMutableSet setWithCapacity: [self count]];
+
+  while ((object = [e nextObject]) != nil)
+    {
+      id result = [object valueForKey: key];
+
+      if (result == nil)
+        continue;
+
+      [results addObject: result];
+    }
+  return results;
+}
+
+// Key-Value Observing Support
+/*
+- addObserver:forKeyPath:options:context:
+- removeObserver:forKeyPath:
+- removeObserver:forKeyPath:context:
+*/
+
+- (NSUInteger)_countForObject: (id)object  // required override...
+{
+  return 1;
+}
+
+// Comparing Sets
+- (BOOL) isEqualToOrderedSet: (NSOrderedSet *)aSet
+{
+  if ([self count] == 0 &&
+      [aSet count] == 0)
+    return YES;
+  
+  if (self == aSet)
+    return YES;
+  
+  if ([self count] != [aSet count])
+    return NO;
+
+  // if they are equal, then this set will be a subset of aSet.
+  return [self isSubsetOfOrderedSet: aSet];
+}
+
+- (BOOL) isEqual: (id)other
+{
+  if ([other isKindOfClass: [NSOrderedSet class]])
+    return [self isEqualToOrderedSet: other];
+  return NO;
+}
+
+// Set operations
+- (BOOL) intersectsOrderedSet: (NSOrderedSet *)otherSet
+{
+  id	o = nil, e = nil;
+
+  // -1. If this set is empty, this method should return NO.
+  if ([self count] == 0)
+    return NO;
+
+  // 0. Loop for all members in otherSet
+  e = [otherSet objectEnumerator];
+  while ((o = [e nextObject])) // 1. pick a member from otherSet.
+    {
+      if ([self containsObject: o])    // 2. check the member is in this set(self).
+        return YES;
+    }
+  return NO;
+}
+
+- (BOOL) intersectsSet: (NSSet *)otherSet
+{
+  id	o = nil, e = nil;
+
+  // -1. If this set is empty, this method should return NO.
+  if ([self count] == 0)
+    return NO;
+
+  // 0. Loop for all members in otherSet
+  e = [otherSet objectEnumerator];
+  while ((o = [e nextObject])) // 1. pick a member from otherSet.
+    {
+      if ([self containsObject: o])    // 2. check the member is in this set(self).
+        return YES;
+    }
+  return NO;
+}
+
+- (BOOL) isSubsetOfOrderedSet: (NSOrderedSet *)otherSet
+{
+  id    f = nil;
+  NSUInteger s = 0, l = [self count], i = 0;
+  
+  // -1. If this set is empty, this method should return YES.
+  if (l == 0)
+    return YES;
+
+  // If count of set is more than otherSet it's not a subset
+  if (l > [otherSet count])
+    return NO;
+
+  // Find the first object's index in otherset, if not found
+  // it's not a subset...
+  f = [self firstObject];
+  s = [otherSet indexOfObject: f];
+  if(s == NSNotFound)
+    return NO;
+
+  for(i = 0; i < l; i++)
+    {
+      NSUInteger j = s + i;
+      id oo = [otherSet objectAtIndex: j];
+      id o = [self objectAtIndex: i];
+
+      if([o isEqual: oo] == NO)
+	{
+	  return NO;
+	}
+    }
+  
+  return YES; // if all members are in set.
+}
+
+- (BOOL) isSubsetOfSet:(NSSet *)otherSet
+{
+    id	o = nil, e = nil;
+
+  // -1. If this set is empty, this method should return NO.
+  if ([self count] == 0)
```

Comment:
Shouldn't the implementation here be similar to the one above?

Reply:
Why did you mark this as resolved as the code still is clearly wrong and different from the ordered case? If self is an empty set it should be a subset and the rest of the code checks for super set not subset.